src/machine/machined-varlink.c

```diff
@@ -408,11 +460,16 @@ static int list_machine_one(sd_varlink *link, Machine *m, bool more) {
                         SD_JSON_BUILD_PAIR_CONDITION(dual_timestamp_is_set(&m->timestamp), "timestamp", JSON_BUILD_DUAL_TIMESTAMP(&m->timestamp)),
                         SD_JSON_BUILD_PAIR_CONDITION(m->vsock_cid != VMADDR_CID_ANY, "vSockCid", SD_JSON_BUILD_UNSIGNED(m->vsock_cid)),
                         JSON_BUILD_PAIR_STRING_NON_EMPTY("sshAddress", m->ssh_address),
-                        JSON_BUILD_PAIR_STRING_NON_EMPTY("sshPrivateKeyPath", m->ssh_private_key_path));
+                        JSON_BUILD_PAIR_STRING_NON_EMPTY("sshPrivateKeyPath", m->ssh_private_key_path),
+                        /* TODO(ikruglov): how do we want to output addresses? Currently JSON_BUILD_IN_ADDR() output address like [ 192, 168, 0, 1 ].
+                         * Are we fine with an array of arrays? Or we want array of strings? Or something else? */
```

Comment:
do it like networkd does it.

Reply:
networkd outputs single address per item if I read the code correctly. I have an array of addresses per machine. So, it will be an array of arrays. Are you fine with it?

Reply:
i certainly am. not entirely sure sd-varlink can actually express that that in the IDL though...

Reply:
but it should be an array of structs of family + array of IP address bytes, no?

Reply:
I mean:

```
{
  ...,
  "Addresses" [
    [ 192, 168, 0, 1],
    [ 192, 168, 0, 2],
  ],
  ...
}
`

Reply:
well, but on networkd we always encode it as family + address, where family is one of AF_INET or AF_INET6